reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/MismatchedLaterality.java

```diff
@@ -3,199 +3,126 @@
+            // Process lateralisable Concepts already added to 723264001 |Lateralisable body structure reference set|.
+            String byMembership = "^ 723264001";
+            Set<Concept> alreadyMembers = getConceptsByECL(byMembership);
+
+            reportOddNumberOfLateralisedChildren(alreadyMembers, true);
+        }
+    }
+
+    private Set<Concept> getConceptsByECL(String... eclStatements) throws TermServerScriptException {
+        Set<Concept> concepts = new HashSet<>();
+        for (String eclStatement : eclStatements) {
+            if (jobRun.getParamBoolean("legacy")) {
+                concepts.addAll(findConcepts(eclStatement));
+            } else {
+                concepts.addAll(findConceptsWithoutEffectiveTime(eclStatement));
+            }
+        }
+
+        return concepts;
+    }
+
+    private void reportOddNumberOfLateralisedChildren(Set<Concept> lateralisableConcepts, boolean memberOfLateralisableReferenceSet) throws TermServerScriptException {
+        int counter = 0;
+        int size = lateralisableConcepts.size();
+        String isMember = memberOfLateralisableReferenceSet ? "Y" : "N";
+        info(String.format("%d lateralisable Concepts will be checked they do not have an odd number of children.", size));
+        for (Concept lateralisableConcept : lateralisableConcepts) {
+            counter++;
+            info(String.format("Processing %d/%d lateralisable Concepts with membership of 723264001 being %s.", counter, size, isMember));
+            if (!lateralisableConcept.isActive()) {
+                report(PRIMARY_REPORT, lateralisableConcept.getConceptId(), lateralisableConcept.getFsn(), lateralisableConcept.getSemTag(), isMember, "Required", "Concept is inactive and should be removed from reference set.");
+                continue;
+            }
+
+            Collection<Concept> lateralisedChildren = findConcepts(String.format("<! %s : 272741003 |Laterality| = (7771000 |left| OR 24028007 |right| OR 51440002 |right and left|)", lateralisableConcept.getConceptId()));
+            if (lateralisedChildren.isEmpty()) {
+                report(PRIMARY_REPORT, lateralisableConcept.getConceptId(), lateralisableConcept.getFsn(), lateralisableConcept.getSemTag(), isMember, "Not required", "No lateralisable children.");
+                continue;
+            }
+
+            int lateralised = lateralisedChildren.size();
+            if (lateralised == 1) {
+                report(PRIMARY_REPORT, lateralisableConcept.getConceptId(), lateralisableConcept.getFsn(), lateralisableConcept.getSemTag(), isMember, "Required", "Possibly missing content as only 1 lateralised child.");
+            } else if (lateralised % 2 != 0) {
+                report(PRIMARY_REPORT, lateralisableConcept.getConceptId(), lateralisableConcept.getFsn(), lateralisableConcept.getSemTag(), isMember, "Required", String.format("Possibly missing content as only %d lateralised children.", lateralised));
+            }
```

Comment:
In 'legacy' mode, I haven't actually had this report complete because 20K Concepts are inspected to determine whether they are at fault. I've tried to optimise my ECL as much as possible. Let me know if you can think of anything else I could do. 

For example, if it was possible to specify the minimum number of children, then 20K wouldn't be inspected as we only care about those with an odd number of children.